testutils/clustermanager/gke_test.go

```diff
@@ -162,16 +166,96 @@ func TestGKECheckEnvironment(t *testing.T) {
 			return out, err
 		}
 
-		err := gc.checkEnvironment()
+		err := fgc.checkEnvironment()
 		var clusterGot *string
-		if nil != gc.Cluster {
-			clusterGot = &gc.Cluster.Name
+		if nil != fgc.Cluster {
+			clusterGot = &fgc.Cluster.Name
 		}
 
-		if !reflect.DeepEqual(err, data.expErr) || !reflect.DeepEqual(gc.Project, data.expProj) || !reflect.DeepEqual(clusterGot, data.expCluster) {
+		if !reflect.DeepEqual(err, data.expErr) || !reflect.DeepEqual(fgc.Project, data.expProj) || !reflect.DeepEqual(clusterGot, data.expCluster) {
 			t.Errorf("check environment with:\n\tkubectl output: '%s'\n\t\terror: '%v'\n\tgcloud output: '%s'\n\t\t"+
 				"error: '%v'\nwant: project - '%v', cluster - '%v', err - '%v'\ngot: project - '%v', cluster - '%v', err - '%v'",
-				data.kubectlOut, data.kubectlErr, data.gcloudOut, data.gcloudErr, data.expProj, data.expCluster, data.expErr, gc.Project, gc.Cluster, err)
+				data.kubectlOut, data.kubectlErr, data.gcloudOut, data.gcloudErr, data.expProj, data.expCluster, data.expErr, fgc.Project, fgc.Cluster, err)
+		}
+	}
+}
+
+func TestAcquire(t *testing.T) {
+	fakeClusterName := "kpkg-e2e-cls-1234"
+	fakeBuildID := "1234"
+	datas := []struct {
+		existCluster       *container.Cluster
+		regionStates       map[string]string
+		expClusterName     string
+		expClusterLocation string
+		expErr             error
+	}{
+		{
+			// cluster already found
+			&container.Cluster{
+				Name:     "customcluster",
+				Location: "us-central1",
+			}, map[string]string{}, "customcluster", "us-central1", nil,
+		}, {
+			// cluster creation succeeded
+			nil, map[string]string{}, fakeClusterName, "us-central1", nil,
+		}, {
+			// cluster creation succeeded retry
+			nil, map[string]string{"us-central1": "PROVISIONING"}, fakeClusterName, "us-west1", nil,
+		}, {
+			// cluster creation failed all retry
+			nil, map[string]string{"us-central1": "PROVISIONING", "us-west1": "PROVISIONING", "us-east1": "PROVISIONING"},
+			"", "", fmt.Errorf("timed out waiting for cluster creation"),
+		}, {
+			// cluster creation went bad state
+			nil, map[string]string{"us-central1": "BAD", "us-west1": "BAD", "us-east1": "BAD"}, "", "", fmt.Errorf("cluster in bad state: 'BAD'"),
+		},
+	}
+
+	// mock GetOSEnv for testing
+	oldFunc := common.GetOSEnv
+	// mock timeout so it doesn't run forever
+	oldTimeout := creationTimeout
+	creationTimeout = 100 * time.Millisecond
+	defer func() {
+		// restore
+		common.GetOSEnv = oldFunc
+		creationTimeout = oldTimeout
+	}()
+
+	for _, data := range datas {
+		common.GetOSEnv = func(key string) string {
+			switch key {
+			case "BUILD_NUMBER":
+				return fakeBuildID
+			case "PROW_JOB_ID": // needed to mock IsProw()
+				return "jobid"
+			}
+			return oldFunc(key)
```

Comment:
We're missing the test cases for setting the region and backup regions using env vars.

Reply:
testing for `setup` function is intentionally not included in this PR, as I don't want to increase the size of current one. Will create a PR for testing `setup` and another one for `delete` once this one is merged

Reply:
/lgtm
/approve